openBCISample.js

```diff
@@ -31,30 +31,39 @@ var sampleModule = {
   *                  calling OpenBCIConstans.channelSettingsArrayInit(). The most important rule here is that it is
   *                  Array of objects that have key-value pair {gain:NUMBER}
   * @param convertAuxToAccel (optional) {Boolean} - Do you want to convert to g's? (Defaults to true)
-  * @returns {Promise} - A sample object
   */
   parseRawPacketStandard: (dataBuf, channelSettingsArray, convertAuxToAccel) => {
     const defaultChannelSettingsArray = k.channelSettingsArrayInit(k.OBCINumberOfChannelsDefault);
+    if (k.isUndefined(convertAuxToAccel) && k.isUndefined(channelSettingsArray)) {
+      // Load default channel settings
+      channelSettingsArray = defaultChannelSettingsArray;
+      // Default for convert Aux to Accel
+      convertAuxToAccel = true;
+    } else if (k.isUndefined(convertAuxToAccel)) {
+      // Default for convert Aux to Accel
+      convertAuxToAccel = true;
+    }
 
-    return new Promise((resolve, reject) => {
-      if (dataBuf === undefined || dataBuf === null) reject('Error [parseRawPacket]: dataBuf must be defined.');
-      // Verify proper start byte
-      if (dataBuf[0] !== k.OBCIByteStart) reject(`Error [parseRawPacket]: Invalid start byte of ${dataBuf[0].toString(16)} expected ${k.OBCIByteStart.toString(16)}`);
-      // channelSettingsArray is optional, defaults to CHANNEL_SETTINGS_ARRAY_DEFAULT
-      channelSettingsArray = channelSettingsArray || defaultChannelSettingsArray;
-      // By default convert to g's
-      if (convertAuxToAccel === undefined || convertAuxToAccel === null) convertAuxToAccel = true;
-
-      if (convertAuxToAccel) {
-        parsePacketStandardAccel(dataBuf, channelSettingsArray).then(sampleObject => {
-          resolve(sampleObject);
-        }).catch(err => reject(err));
-      } else {
-        parsePacketStandardRawAux(dataBuf, channelSettingsArray).then(sampleObject => {
-          resolve(sampleObject);
-        }).catch(err => reject(err));
-      }
-    });
+    // Check to make sure data is not null.
+    if (k.isUndefined(dataBuf) || k.isNull(dataBuf) || k.isNumber(dataBuf)) {
```

Comment:
They are helper functions defined in the constants file. I just added tests for them too.